src/main/java/seedu/contax/logic/commands/BatchCommand.java

```diff
@@ -37,40 +40,63 @@
             + PREFIX_SEARCH_TYPE + "phone "
             + PREFIX_EQUALS + "123 ";
 
+    private final Logger logger = LogsCenter.getLogger(BatchCommand.class);
+
     private final String commandInput;
     private final SearchType searchType;
-    private final String userInput;
+    private final String equalsText;
+    private final String startText;
+    private final String endText;
 
     /**
      * Creates an BatchCommand base on specified {@code commandInput} {@code searchType} and {@code userInput}.
      *
      * @param commandInput              details to word of command
      * @param searchType                search type of field apply matcher
-     * @param userInput                 regex provided by user
+     * @param equalsText                equals provided by user
+     * @param startText                equals provided by user
+     * @param endText                equals provided by user
      */
-    public BatchCommand(String commandInput, SearchType searchType, String userInput) {
+    public BatchCommand(String commandInput, SearchType searchType,
+                        String equalsText, String startText, String endText) {
```

Comment:
It would be more appropriate to pass in a single text search term, and an Enum search mode, since you are only using 1 text field at at one time. The empty string check is very fragile.

Reply:
thx will changed